### PR TITLE
[GPU] Fix NaNs in single-token SDPA with conditional attention mask clamp

### PR DESCRIPTION
**Pull request by andrew-k-park**
### Details:
 - Unlike the multi-token SDPA opt kernel, the single-token SDPA opt kernel sets the softmax accumulator type to FP32, so it does not perform an explicit clamp
 - However, in an extreme case where the attention mask is FP16 and only the first element is zero  while all others are -inf, the kernel can produce NaNs before the softmax operation
 - Added a JIT constant to check when the attention mask dtype differs from the softmax accumulator type
 - Updated the single-token SDPA opt kernel to conditionally clamp the attention mask preventing NaN generation

#### The code and line that caused this issue (if it is not changed directly)
https://github.com/openvinotoolkit/openvino/blob/1172c680e8c1afd572c30534a21849d102566321/src/plugins/intel_gpu/src/graph/impls/ocl_v2/sdpa_opt.cl#L439

#### Reproduction step and snapshot (if applicable. Do not attach for customer model)
 - https://github.com/andrew-k-park/whisper-ovep-python-static/blob/use_ov_native_api/whisper_openvino.py
 - python whisper_openvino.py --model_type base --device GPU how_are_you_doing_today.wav

#### Problematic graph
image
image
image

#### Checklist
 - [x] Is it a proper fix? (not a workaround)
 - [x] Did you include test case for this fix, if necessary?
 - [x] Did you review existing test that can be extended to cover this scenario? Which test did you review?
   - https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_gpu/tests/unit/test_cases/sdpa_gpu_test.cpp

### Tickets:
 - [CVS-176516](https://jira.devtools.intel.com/browse/CVS-176516)

### CODE REVIEW

**Review by e-ddykim — APPROVED** [review]

File: src/plugins/intel_gpu/tests/unit/test_cases/sdpa_gpu_test.cpp [hunk 1]
```diff
@@ -282,4 +282,82 @@ TEST_P(sdpa_gpu_test, basic_caching) {
     execute(p, true);
 }
 #endif
+
+TEST(sdpa_gpu_custom, mask_single_valid_rest_inf) {
```
Comment (isanghao) [inline src/plugins/intel_gpu/tests/unit/test_cases/sdpa_gpu_test.cpp]: nit: this test name is difficult to understand. does it reflect your intention?

Comment (andrew-k-park) [reply src/plugins/intel_gpu/tests/unit/test_cases/sdpa_gpu_test.cpp]: updated
